src/main/java/com/google/cloud/spanner/jdbc/AbstractJdbcWrapper.java

```diff
@@ -69,7 +70,74 @@ static int extractColumnType(Type type) {
     }
   }
 
-  /** Extract Spanner type name from {@link java.sql.Types} code. */
+  static String getSpannerTypeName(Type type, Dialect dialect) {
```

Comment:
My plan was to add that to the Java client library instead of directly in the JDBC driver, so we can use it in more places. I've expedited that and added a PR for it here: https://github.com/googleapis/java-spanner/pull/2763